icu4c/source/common/uresbund.cpp

```diff
@@ -3161,31 +3165,31 @@ ures_getFunctionalEquivalent(char *result, int32_t resultCapacity,
         subStatus = U_ZERO_ERROR;
 
         if (res != nullptr) {
-            uprv_strcpy(found, ures_getLocaleByType(res, ULOC_VALID_LOCALE, &subStatus));
+            found.clear().append(ures_getLocaleByType(res, ULOC_VALID_LOCALE, &subStatus), subStatus);
         }
 
-        if (uprv_strcmp(found, parent) != 0) {
-            uprv_strcpy(parent, found);
+        if (found != parent.toStringPiece()) {
```

Comment:
Can't we simply do
```
    if (found != parent) {
```
?

Reply:
> Can't we simply do
> 
> ```
>     if (found != parent) {
> ```
> 
> ?

No, that is currently not possible, but I think that it should be, which is why I sent to PR #2764 for review last week to make that possible.